src/wasm_emulation/channel.rs

```diff
@@ -18,18 +14,22 @@ pub struct RemoteChannel {
     pub rt: Handle,
     pub channel: Channel,
     pub pub_address_prefix: String,
+    // For caching
+    pub chain_id: String,
 }
 
 impl RemoteChannel {
     pub fn new(
         rt: &Runtime,
         chain: impl Into<ChainInfoOwned>,
-        pub_address_prefix: impl Into<String>,
+        _: impl Into<String>,
```

Comment:
Not related to this task, but I think we can remove this field, or is there any cases where we would want to have different address prefix with chain?

Reply:
don't think so!